pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -370,6 +379,55 @@ func (b *EndpointBuilder) BuildClusterLoadAssignment(endpointIndex *model.Endpoi
 	return l
 }
 
+func (b *EndpointBuilder) generateToServiceWaypoint() []*LocalityEndpoints {
+	if b.proxy.IsWaypointProxy() || b.dir != model.TrafficDirectionOutbound || !features.EnableHBONE {
+		return nil
```

Comment:
Will a gateway return true for 'is waypoint proxy' ? Do we expect Gateways to chain to a waypoint ( and is the waypoint correctly dealing with chaining ) ?

Reply:
A gateway _can_ be a Waypoint. I believe non-waypoint gateways should chain to waypoints. That's what they do today at least. I'm not looking to change that in this PR.